app/views/vendors/_orderitems.html.erb

```diff
@@ -0,0 +1,18 @@
+<h1> <%= filter %> </h1>
+<% @order_items.where(status: filter).each do | order_item | %>
```

Comment:
The filtering generally should not be done within the view - but rather the controller (or even the model). Whatever order items you're looking for should be passed to the view - already filtered